SparkPost/2web2ui
# FE-1153 - Adds new Layout component wrapper for the Matchbox Layout component

[FE-1153](https://jira.int.messagesystems.com/browse/FE-1153)

### What Changed
- Upgrades Matchbox to beta release with the Layout component (see the ticket for more details)
- Adds new `<Layout/>` component wrapper for Matchbox and supporting unit tests

### How To Test
- Try out the new component! With Hibana enabled, try it out:
```
import React from 'react';
import { Layout, Panel } from 'src/components/matchbox';

function MyComponent() {
  return (
    <Layout>
      <Layout.Section annotated>
        <Layout.SectionTitle as="h2">
          Domain Status
        </Layout.SectionTitle>
      </Layout.Section>
      <Layout.Section>
        <Panel>
        </Panel>
      </Layout.Section>
    </Layout>
  )
}
```
- Make sure it throws an error when the OG theme is enabled

### To Do
